src/main/java/spireMapOverhaul/zones/heavenlyClouds/HeavenlyCloudsZone.java

```diff
@@ -0,0 +1,60 @@
+package spireMapOverhaul.zones.heavenlyClouds;
+
+import basemod.helpers.CardModifierManager;
+import com.badlogic.gdx.graphics.Color;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import spireMapOverhaul.abstracts.AbstractZone;
+import spireMapOverhaul.util.Wiz;
+import spireMapOverhaul.zoneInterfaces.CombatModifyingZone;
+import spireMapOverhaul.zoneInterfaces.RewardModifyingZone;
+
+import java.util.ArrayList;
+
+import static spireMapOverhaul.util.Wiz.atb;
+
+public class HeavenlyCloudsZone extends AbstractZone implements CombatModifyingZone, RewardModifyingZone {
+    public static final String ID = "HeavenlyClouds";
+    private static final float HEALTH_MODIFIER = 0.70f;
+    public HeavenlyCloudsZone() {
+        super(ID, Icons.MONSTER);
+        this.width = 3;
+        this.maxHeight = 4;
+        this.height = 4;
+        this.maxHeight = 5;
+    }
+
+    @Override
+    public AbstractZone copy() {
+        return new HeavenlyCloudsZone();
+    }
+
+    @Override
+    public Color getColor() {
+        return Color.SKY.cpy();
+    }
+
+    @Override
+    public void atBattleStart() {
+        int flightAmount = 3;
+        if (AbstractDungeon.ascensionLevel >= 19) {
+            flightAmount = 4;
+        }
+        for (AbstractMonster mo : Wiz.getEnemies()) {
+            atb(new ApplyPowerAction(mo, mo, new HeavenlyFlightPower(mo, flightAmount), flightAmount));
+            mo.currentHealth = (int)((float)mo.maxHealth * HEALTH_MODIFIER);
```

Comment:
If something causes enemies to start with lower current health than max health (e.g. Preserved Insect), this will wipe that out, right? I believe the zone hooks mostly happen after the equivalent relic hooks.

I think it would be better to apply the percentage health reduction to both current and max health, instead of setting current to a percentage of max.
```suggestion
            mo.currentHealth = (int)((float)mo.currentHealth * HEALTH_MODIFIER);
```

Reply:
added this change so it works with preserved insect